monitoring/src/test/scala/akkaviz/server/WebSocketReplTest.scala

```diff
@@ -0,0 +1,135 @@
+package akkaviz.server
+
+import java.io.{InputStream, OutputStream}
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.ws.{BinaryMessage, Message, TextMessage}
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Keep, Sink, Source}
+import akka.stream.testkit.TestSubscriber.OnNext
+import akka.stream.testkit.scaladsl.{TestSink, TestSource}
+import akka.util.ByteString
+import ammonite.ops.Path
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{FunSuite, Matchers}
+
+import scala.concurrent.duration._
+
+class WebSocketReplTest extends FunSuite with ScalaFutures with Matchers {
+
+  private[this] implicit val system = ActorSystem()
+  private[this] implicit val materializer = ActorMaterializer()
+
+  override implicit val patienceConfig: PatienceConfig = PatienceConfig(10.seconds)
+
+  private[this] def createRepl(fn: Function2[InputStream, OutputStream, Unit]) = new WebSocketRepl {
+    override protected def startReplProcessing(
+      in: InputStream, out: OutputStream, err: OutputStream, homePath: Path
+    ): Unit = {
+      fn(in, out)
+    }
+  }
+
+  private[this] def wrongMsg = fail("Response should be BinaryMessage.Strict")
+
+  private[this] val pingPongAndClose: Function2[InputStream, OutputStream, Unit] = {
+    (in, out) =>
+      val ab = new Array[Byte](4)
+      in.read(ab)
+      ByteString(ab).utf8String shouldBe "ping"
+      out.write("pong".getBytes)
+      in.close()
+      out.flush()
+      out.close()
+  }
+
+  private[this] def expectString(string: String)(chunks: Seq[Message]) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use : Unit = for procedures](https://www.codacy.com/app/lustefaniak/akka-viz/file/2329347419/issues/source?bid=3121282&fileBranchId=3206949#l46)